Shopify/cli
# Send the command generate as scaffold in analytics events

Addresses: https://github.com/Shopify/cli/pull/376#discussion_r961440781

### WHY are these changes introduced?

This PR addresses @shauns's point in [this comment](https://github.com/Shopify/cli/pull/376#discussion_r961440781). With the recent rename of `scaffold` to `generate` we'd send events with `command: generate` resulting in two types of events that represent the same thing but with different names. This is prone to confusion in the future because any queries and dashboards will have to take this into account.

### WHAT is this pull request doing?
I'm adjusting the analytics logic to allow commands to override the command name. 

### How to test your changes?
Run `yarn shopify app generate extension --path ./fixtures/app --verbose` and you should see in the analytics event that was sent that `command_name` has `scaffold` as value.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
